File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/impl/TimerDao.java
package com.dkellycollins.triggerfinger.data.daos.impl;

import com.dkellycollins.triggerfinger.data.daos.ITimerDao;
import com.dkellycollins.triggerfinger.data.entity.ITimer;
import com.dkellycollins.triggerfinger.data.entity.impl.Timer;

import java.util.HashMap;

public class TimerDao extends BaseEntityDao<Timer> implements ITimerDao {

    public TimerDao() {
        super("TimerDao");
    }

    @Override
    public Iterable<ITimer> retrieve() {
        return (Iterable<ITimer>) (Iterable<?>) _store.values();
    }

    @Override
    public ITimer retrieve(int id) {
        return _store.get(id);
    }

    @Override
    public int create(int setTime, int currentTime, boolean isRunning) {
        int id = getNextId();
        Timer timer = new Timer(id, setTime, currentTime, isRunning);

        _store.put(id, timer);

        return id;
    }

    @Override
    public void update(int id, int currentTime, boolean isRunning) {
        Timer timer = _store.get(id);

        timer.setCurrentTime(currentTime);
        timer.setRunning(isRunning);
    }

    @Override
    public void delete(int id) {
        _store.remove(id);
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/impl/TimerDao.kt
package com.dkellycollins.triggerfinger.data.daos.impl

import com.dkellycollins.triggerfinger.data.daos.ITimerDao
import com.dkellycollins.triggerfinger.data.entity.ITimer
import com.dkellycollins.triggerfinger.data.entity.impl.Timer

import java.util.HashMap

class TimerDao : BaseEntityDao<Timer>("TimerDao"), ITimerDao {

    override fun retrieve(): Iterable<ITimer> {
        return _store.values
    }

    override fun retrieve(id: Int): ITimer {
        return _store[id]
    }

    override fun create(setTime: Int, currentTime: Int, isRunning: Boolean): Int {
        val id = nextId
        val timer = Timer(id, setTime, currentTime, isRunning)

        _store.put(id, timer)

        return id
    }

    override fun update(id: Int, currentTime: Int, isRunning: Boolean) {
        val timer = _store[id]

        timer.currentTime = currentTime
        timer.isRunning = isRunning
    }

    override fun delete(id: Int) {
        _store.remove(id)
    }
}
